Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,257 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
+              )
+            }),
+            rightParen: ")"
+          ))
+        })
+      )
+      associatedMembers.append(encodeForwarder)
+    }
+    if conformingToDecodable {
+      let decodeForwarder = InitializerDeclSyntax(
+        signature: .init(
+          parameterClause: .init(parametersBuilder: {
+            .init(
+              firstName: "from",
+              secondName: "decoder",
+              type: SomeOrAnyTypeSyntax(
+                someOrAnySpecifier: .keyword(.`any`),
+                constraint: IdentifierTypeSyntax(name: "Decoder")
+              )
+            )
+          }),
+          effectSpecifiers: throwsEffect
+        ),
+        body: .init(statementsBuilder: {
+          InfixOperatorExprSyntax(
+            leftOperand: MemberAccessExprSyntax(
+              base: DeclReferenceExprSyntax(baseName: .keyword(.`self`)),
+              declName: DeclReferenceExprSyntax(baseName: storageName)
+            ),
+            operator: AssignmentExprSyntax(),
+            rightOperand: TryExprSyntax(expression: FunctionCallExprSyntax(
+              calledExpression: MemberAccessExprSyntax(
+                declName: .init(baseName: .keyword(.`init`))
+              ),
+              leftParen: "(",
+              arguments: .init(itemsBuilder: {
+                .init(
+                  label: "from",
+                  expression: DeclReferenceExprSyntax(baseName: "decoder")
+                )
+              }),
+              rightParen: ")"
+            ))
+          )
+        })
+      )
+      associatedMembers.append(decodeForwarder)
+    }
+  }
+  
+  private func createDerivedStorageTypeDecl() -> StorageTypeAndAssociatedMembers {
+    var members = appliedStructValidVarDecls.map {
+      return MemberBlockItemSyntax(decl: $0)
     }
     
+    var protocols = appliedStructDecl.collectAutoSynthesizingProtocolConformance()
+    var associatedMembers = [any DeclSyntaxProtocol]()
+    applyEquatableWorkaroundIfNeeded(members: &members,
+                                     protocols: &protocols,
+                                     associatedMembers: &associatedMembers)
+    forwardCodingKeysForDerivedStorageIfNeeded(
+        members: &members,
+        protocols: &protocols,
+        associatedMembers: &associatedMembers
+    )
     
     let inheritance = InheritanceClauseSyntax {
       InheritedTypeListSyntax {
         InheritedTypeSyntax(type: CopyOnWriteStorage.type)
       }
       // Equatable, Hashable, Codeable
       InheritedTypeListSyntax(
-        appliedStructDecl.collectAutoSynthesizingProtocolConformance()
+        protocols
       )
     }
     
-    let typeName: TokenSyntax = "_$COWStorage"
-    
-    return StructDeclSyntax(
-      name: typeName,
-      inheritanceClause: inheritance
-    ) {
-      MemberBlockItemListSyntax(members)
-    }
+    return (
+      StructDeclSyntax(
+        name: defaultStorageTypeName,
```

Comment:
I think it will enhance cohesion to move `let defaultStorageTypeName: TokenSyntax = "_$COWStorage"` to (the start or the end of, both OK) this file and make the declaration `private`.